cli/src/torii/routing.rs

```diff
@@ -170,32 +170,39 @@ pub(crate) async fn handle_queries(
 }
 
 fn apply_sorting_and_pagination(
-    mut vec_of_val: Vec<Value>,
+    vec_of_val: Vec<Value>,
     sorting: &Sorting,
     pagination: Pagination,
 ) -> Vec<Value> {
+    let mut iter: Box<dyn Iterator<Item = Value>> = Box::new(vec_of_val.into_iter());
```

Comment:
You need this binding to get rid of the `mut` in the signature?

Reply:
I tried to use if-else to avoid `collect` calls and to get an iterator, but the iterators of both branches didn't match. That's why I used an object trait.